src/libratbag.c

```diff
@@ -1004,7 +999,10 @@ ratbag_resolution_set_dpi(struct ratbag_resolution *resolution,
 {
 	struct ratbag_profile *profile = resolution->profile;
 
-	if (!resolution_has_dpi(resolution, dpi))
+	if (resolution->is_disabled) {
+		if (!ratbag_resolution_has_capability(resolution, RATBAG_RESOLUTION_CAP_DISABLE))
+			return RATBAG_ERROR_CAPABILITY;
```

Comment:
Hm, I don't think there is a need for any of these checks at all.

As is, this checks for disabling capability in DPI setting, an unrelated place.

If this is changed to check if resolution is simply disabled, then still I don't see a need for this, as I think it should be possible to change the DPI of a resolution even if it's disabled.